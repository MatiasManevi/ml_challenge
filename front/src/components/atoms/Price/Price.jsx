import React from 'react';

import './Price.scss';

const Price = ({ price, small = true, freeShipping = false }) => {
	return (
		<div className="price">
			<p className={small ? 'price--small' : 'price--big'}>
				{price.currency} {price.amount}
				<sup
					className={
						'price__cents ' + (small ? 'price__cents--small' : 'price__cents--big')
					}
				>
					{price.decimals}
				</sup>
			</p>
			{freeShipping && <img className="price__shipping" src="ic_shipping.png" />}
		</div>
	);
};

export default Price;
