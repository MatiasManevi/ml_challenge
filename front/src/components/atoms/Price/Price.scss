@import 'assets/scss/variables';

.price {
	margin-bottom: $space-32;
	display: flex;
	align-items: center;
}

.price--small {
	font-size: $font-size-24;
}

.price--big {
	font-size: $font-size-46;
}

.price__shipping {
	margin-left: $space-4;
}

.price__cents {
	margin-left: $space-4;
}

.price__cents--small {
	font-size: $font-size-14;
}

.price__cents--big {
	font-size: $font-size-24;
}
