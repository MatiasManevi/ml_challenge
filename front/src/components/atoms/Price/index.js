export { default } from 'components/atoms/Price/Price';
