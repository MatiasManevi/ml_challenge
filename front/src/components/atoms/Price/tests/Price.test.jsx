import React from 'react';
import { shallow } from 'enzyme';

import Price from '../Price';

describe('Price', () => {
	const price = 2000;

	it('should match snapshot', () => {
		const wrapper = shallow(<Price price={price} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should use the right class', () => {
		const wrapper = shallow(<Price price={price} small={true} />);
		expect(wrapper.find('p').props().className).toBe('price--small');
	});

	it('should use shipping icon', () => {
		const wrapper = shallow(<Price price={price} freeShipping={true} />);
		expect(wrapper.find('img').length).toBe(1);
	});
});
