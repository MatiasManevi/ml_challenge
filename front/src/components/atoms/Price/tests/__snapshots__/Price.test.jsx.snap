// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price should match snapshot 1`] = `
<div
  className="price"
>
  <p
    className="price--small"
  >
     
    <sup
      className="price__cents price__cents--small"
    />
  </p>
</div>
`;
